kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableWriteWithCrcSuite.scala

```diff
@@ -0,0 +1,115 @@
+package io.delta.kernel.defaults
+import io.delta.kernel.data.Row
+import io.delta.kernel.{Transaction, TransactionCommitResult}
+import io.delta.kernel.defaults.utils.TestRow
+import io.delta.kernel.engine.Engine
+import io.delta.kernel.hook.PostCommitHook.PostCommitHookType
+import io.delta.kernel.internal.TransactionImpl
+import io.delta.kernel.internal.actions.Metadata
+import io.delta.kernel.internal.checksum.CRCInfo
+import io.delta.kernel.internal.util.Utils.singletonCloseableIterator
+import io.delta.kernel.types.StructType
+import io.delta.kernel.utils.{CloseableIterable, FileStatus}
+
+import java.io.File
+import java.nio.file.Files
+import java.util.{Locale, Optional}
+import scala.collection.immutable.Seq
+import scala.jdk.CollectionConverters.{asScalaBufferConverter, asScalaSetConverter}
+import scala.language.implicitConversions
+
+class DeltaTableWriteWithCrcSuite extends DeltaTableWritesSuite {
+
+  implicit class TransactionOps(txn: Transaction) {
```

Comment:
why do you need this class? can't you just move this implementation of `commitAndGenerateCrc` into the `override def commitTransaction(` below? Please let me know if I'm missing something.